scanner/src/main/kotlin/scanners/Askalono.kt

```diff
@@ -49,9 +49,9 @@ import okhttp3.Request
 
 import okio.Okio
 
-class Askalono(config: ScannerConfiguration) : LocalScanner(config) {
-    class Factory : AbstractScannerFactory<Askalono>() {
-        override fun create(config: ScannerConfiguration) = Askalono(config)
+class Askalono(name: String, config: ScannerConfiguration) : LocalScanner(name, config) {
```

Comment:
Maybe. I didn't fully think this through to be honest but just wanted to stay flexible. A possible use-case coming to my mind now would be to use the same scanner but with different configurations, and referring to these under different names.

Reply:
Also, for now that's in-line with like we already do it for `PackageManger`.

Reply:
I'd rather have hard coded names in `PackageManager`, too, but let's have it consistent for now and refactor later.

Reply:
BTW, I now remember the original reason why I didn't hard-code the name in the PackageManager: The PackageManagerFactory also needs to know the PackageManager name in order to have it displayed in the CLI help through JCommander. As a factory knows about its package manager, but a package manager knows nothing about its factory, and I didn't want to hard-code the name twice, I decided to hard-code it only in the factory, and pass the name to the package manager. I tried all sorts of tricks (with `reified T` etc.) to make a factory statically get a package manager's name, but I did not succeed and finally gave up in favor of this solution.

Reply:
What about using a constant for the name, and using it for initializing both, the factory and the package manager?

```patch
index 4cbe3cb9..7f22d75f 100644
--- a/analyzer/src/main/kotlin/managers/NPM.kt
+++ b/analyzer/src/main/kotlin/managers/NPM.kt
@@ -67,8 +67,8 @@ import okhttp3.Request
 /**
  * The Node package manager for JavaScript, see https://www.npmjs.com/.
  */
-open class NPM(name: String, analyzerConfig: AnalyzerConfiguration, repoConfig: RepositoryConfiguration) :
-        PackageManager(name, analyzerConfig, repoConfig), CommandLineTool {
+open class NPM(analyzerConfig: AnalyzerConfiguration, repoConfig: RepositoryConfiguration) :
+        PackageManager(NAME, analyzerConfig, repoConfig), CommandLineTool {
     companion object {
         /**
          * Expand NPM shortcuts for URLs to hosting sites to full URLs so that they can be used in a regular way.
@@ -108,13 +108,15 @@ open class NPM(name: String, analyzerConfig: AnalyzerConfiguration, repoConfig:
                 url
             }
         }
+
+        const val NAME = "NPM"
     }
 
-    class Factory : AbstractPackageManagerFactory<NPM>("NPM") {
+    class Factory : AbstractPackageManagerFactory<NPM>(NAME) {
         override val globsForDefinitionFiles = listOf("package.json")
 
         override fun create(analyzerConfig: AnalyzerConfiguration, repoConfig: RepositoryConfiguration) =
-                NPM(managerName, analyzerConfig, repoConfig)
+                NPM(analyzerConfig, repoConfig)
     }
 
     protected open fun hasLockFile(projectDir: File) = PackageJsonUtils.hasNpmLockFile(projectDir)
```

Reply:
What I don't like so much about this approach is that you have both `NPM.NAME` and `PackageManager.managerName` then, possibly creating some confusion which one to use. You *could* make the latter `protected`, but then again having a `NAME` for a package manager is just a convention, and not enforced by the compiler.